packages/web-components/src/checkbox/checkbox.styles.ts

```diff
@@ -193,4 +193,39 @@ export const styles = css`
   :host([disabled]) .indeterminate-indicator {
     background-color: ${colorNeutralForegroundDisabled};
   }
-`;
+`.withBehaviors(
+  forcedColorsStylesheetBehavior(css`
+    .checked-indicator {
+      color: Highlight;
+    }
+    :host(:active) .indeterminate-indicator,
+    .indeterminate-indicator {
+      background-color: Highlight;
+    }
+    :host([aria-checked='true']) .control,
+    :host([aria-checked='mixed']) .control,
+    :host([aria-checked='true']:hover) .control,
+    :host([aria-checked='mixed']:hover) .control,
+    :host(:active) .control {
```

Comment:
I would use the `:is()` selector here to cut down on the size of this selector rule:

```css
:host(:is([aria-checked='true'], [aria-checked='mixed'], [aria-checked='true']:hover, [aria-checked='mixed']:hover, :active) .control {}
```
(Pseudo-code, please validate :))